Realm/ObjectServerTests/RLMFlexibleSyncServerTests.mm

```diff
@@ -1013,19 +1008,28 @@ - (void)testFlexibleSyncInitialSubscriptionRerunOnOpen {
         [ex fulfill];
     }];
     [self waitForExpectationsWithTimeout:10.0 handler:nil];
+    XCTAssertEqual(openCount, 1);
 
-    [self clearCachedRealms];
-
+    __block RLMRealm *realm;
     XCTestExpectation *ex2 = [self expectationWithDescription:@"download-realm-2"];
     [RLMRealm asyncOpenWithConfiguration:config
                            callbackQueue:dispatch_get_main_queue()
-                                callback:^(RLMRealm *realm, NSError *error) {
+                                callback:^(RLMRealm *r, NSError *error) {
+        realm = r;
         XCTAssertNil(error);
         XCTAssertEqual(realm.subscriptions.count, 2UL);
         CHECK_COUNT(16, Person, realm);
         [ex2 fulfill];
     }];
     [self waitForExpectationsWithTimeout:10.0 handler:nil];
+    XCTAssertEqual(openCount, 2);
+
+    [self dispatchAsyncAndWait:^{
+        [RLMRealm realmWithConfiguration:config error:nil];
+        // Should not have called initial subscriptions despite rerunOnOpen being
```

Comment:
I don't understand this message, in this case initialSubscription should be called? which is correct when compare openCount == 2 on the next line, or am I missing something?

Reply:
The Realm is already open on the main thread, and initialSubscriptions is only supposed to be called when opening the Realm from scratch.